compiler/src/dotty/tools/dotc/semanticdb/PPrint.scala

```diff
@@ -0,0 +1,252 @@
+package dotty.tools.dotc.semanticdb
+
+import dotty.tools.dotc.{semanticdb => s}
+
+import scala.collection.mutable
+import dotty.tools.dotc.semanticdb.Scala3.{_, given}
+import SymbolInformation.Kind._
+
+class SymbolInfomationPrinter (symtab: PrinterSymtab):
+  val notes = InfoNotes()
+  val infoPrinter = InfoPrinter(notes)
+
+  def pprintSymbolInformation(info: SymbolInformation): String =
+    val sb = new StringBuilder()
+    sb.append(info.symbol).append(" => ")
+    sb.append(infoPrinter.pprint(info))
+    sb.toString
+
+  class InfoNotes:
+    private val noteSymtab = mutable.Map[String, SymbolInformation]()
+    def enter(info: SymbolInformation) =
+      if (symtab.info(info.symbol).isEmpty && info.kind != UNKNOWN_KIND)
+        noteSymtab(info.symbol) = info
+
+    def visit(sym: String): SymbolInformation =
+      val symtabInfo = noteSymtab.get(sym).orElse(symtab.info(sym))
```

Comment:
- `noteSymtab` is for symbols that are hard-linked in structural types
- `symtab` is SymbolTable for a TextDocument

We need symbol table, because when we convert `TypeRef` into Semanticdb Type, we have to lookup the actual symbol from sym-linked symbol.